File: live-test/java/src/main/java/demo/GenericSuper.java
// Super-method fixture, generic case: Repo.find overrides BaseRepo.find,
// whose signature is declared with the type parameter T — the super must
// resolve across the generic signature.
package demo;

abstract class BaseRepo<T> {
    abstract T find(int id);
}

class Repo<T> extends BaseRepo<T> {
    @Override
    T find(int id) {
        return null;
    }
}

File: live-test/kotlin/src/main/kotlin/demo/GenericSuper.kt
// Generic super resolution: Repo<T>.find must resolve to BaseRepo<T>.find
// through the type-parameter substitution.
package demo

abstract class BaseRepo<T> {
    abstract fun find(id: Int): T
}

class Repo<T> : BaseRepo<T>() {
    override fun find(id: Int): T {
        throw NotImplementedError()
    }
}
